src/main/java/net/mcreator/ui/component/util/ComponentUtils.java

```diff
@@ -118,4 +119,13 @@ public static JPanel squareAndBorder(Component gor, String text) {
 		return squareAndBorder(gor, (Color) UIManager.get("MCreatorLAF.BRIGHT_COLOR"), text);
 	}
 
+	public static enum Anchor {

```

Comment:
First of all, this Anchor enum is not implemented correctly. It should not be implemented in this Utils class. A better implementation would be inside a more general/abstract class (e.g. GUIComponent or a new AnchorableComponent class) or an even better way is the usage of an interface tbat can then be applied to GUIComponents supportong the feature.

Reply:
I had some trouble finding a good place for it and I'm happy to move it wherever I can't see anything wrong about the enum itself though.

>  A better implementation would be something similar to how creative tabs are handled with blovks and items
Could you elaborate on that? If I understand you correctly you suggest implementing it as a MappableElement!?
Why would that be better?

Reply:
I said the data list (so the mapping) is not needed. I meant the logic behind adding creative tabs to mod elements (with the interface)

Reply:
I'm sorry, I don't how I would use DataListComboBox without a data list provider (something like ElementUtils.loadAllTabs)
I understand that your time is limited but I'd appreciate if you'd be a bit more verbose in your responses.

From what I understood I should do the following:
- Change the type of LabelDialog.anchor from JComboBox to DataListComboBox
- Add ElementUtils.getAnchorPoints to get a List<DataListEntry>

Am I on the right track?

Reply:
No, you're not. I was taliking about the logic inside yhe saving classes only (Block.java, Item.java). How it is saved and how it is used inside the class (the interface only, ICreativeTab)

Reply:
Apparently I bit off more than I can chew.

I have no idea what you are trying to tell me

Reply:
Something like [this interface](https://github.com/MCreator/MCreator/blob/master/src/main/java/net/mcreator/element/types/interfaces/ITabContainedElement.java). However, you have to replace TabEntry by AnchorPoint (an enum you can move to this interface)

Reply:
hope this is what you head in mind